src/index.js

```diff
@@ -14,27 +16,88 @@ const traverse = require('traverse');
 
 const DEFAULT_REPLACEMENT = '--REDACTED--';
 
+/**
+ * Validate serializers.
+ */
+
+function validateSerializers(serializers) {
+  serializers.map(({ path, serializer }) => {
+    if (typeof serializer !== 'function') {
+      throw new TypeError(`Invalid serializer for \`${path}\` path: must be a function`);
+    }
+  });
+}
+
+/**
+ * Compute Mutations
+ */
+
+function computeSerializedChanges(values, serializers) {
+  const changes = {};
+
+  for (const { path, serializer } of serializers) {
+    const value = get(values, path);
+
+    if (value === undefined) {
+      continue;
+    }
+
+    try {
+      changes[path] = serializer(value);
+    } catch (error) {
+      changes[path] = `Anonymize ERROR: Error while applying ${path} serializer`;
+    }
+  }
+
+  return changes;
+}
+
 /**
  * Module exports.
+ *
+ * Example:
+ *
+ * anonymizer({
+ *   whitelist: ['foo']
+ * }, {
+ * replacement,
+ * serializers: [
+ *   { path: 'foo.bar', serializer: () => {} }
+ *  ]
+ * })
  */
 
-module.exports = (
+module.exports.anonymizer = (
   { blacklist = [], whitelist = [] } = {},
-  { replacement = () => DEFAULT_REPLACEMENT, trim = false } = {}
+  { replacement = () => DEFAULT_REPLACEMENT, serializers = [], trim = false } = {}
 ) => {
   const whitelistTerms = whitelist.join('|');
   const whitelistPaths = new RegExp(`^(${whitelistTerms.replace(/\./g, '\\.').replace(/\*/g, '.*')})$`, 'i');
   const blacklistTerms = blacklist.join('|');
   const blacklistPaths = new RegExp(`^(${blacklistTerms.replace(/\./g, '\\.').replace(/\*/g, '.*')})$`, 'i');
 
+  validateSerializers(serializers);
+
   return values => {
     if (!(values instanceof Object)) {
       return values;
     }
 
     const blacklistedKeys = new Set();
+    // JSON.parse(stringify(values)) builds an object copy that isn't an
```

Comment:
Please leave a empty line before starting this comment.

Reply:
Done